austation/code/modules/mining/minebot.dm

```diff
@@ -0,0 +1,155 @@
+
+/obj/structure/marker_beacon/attack_animal(mob/living/simple_animal/M)
+	if (istype(M, /mob/living/simple_animal/hostile/mining_drone))  //Are we a minebot
+		var/mob/living/simple_animal/hostile/mining_drone/user = M
+		if (user.mode == MINEDRONE_COLLECT)	  //Are we in collect mode (in attack mode, we'll just attack it)
+			if (user.beacons < 15)  // Do we have room for more?
+				user.beacons++  //Increase our inventory
+				to_chat(user, "<span class='notice'>You pick up the [src].</span>")
+				to_chat(user, "<span class='notice'>You have a total of [user.beacons] [src]s.</span>")
+				qdel(src)  //Remove the beacon from the floor
+				return
+			else
+				to_chat(user, "<span class='notice'>You already possess 15 [src]s!</span>")
+	..()
+
+
+/////////////////////////  ORES  STUFF  /////////////////////////
+
+
+/mob/living/simple_animal/hostile/mining_drone/Move()
+	..()
+	if (. && mode == MINEDRONE_COLLECT) // if we're collecting ore, we should do it automatically, not by clicking the ores
+		CollectOre()
+
+/mob/living/simple_animal/hostile/mining_drone/DropOre(message = 1)
+	..()
+	SetOffenseBehavior() // prevents the drone from picking up the ores again as it walks away
+	to_chat(src, "<span class='notice'>You resume combat protocols.</span>")
+
+
+/////////////////////////  TECHWEB  &  SPAWNING  /////////////////////////
+
+
+/obj/effect/mob_spawn/minebot  //what comes out of the minebot fab
+	name = "unactivated minebot"
+	desc = "A currently unactivated minebot. After activating, it may assist you in mining ores and fighting wildlife."
+	icon = 'icons/mob/swarmer.dmi'
+	icon_state = "swarmer_unactivated"
+	density = FALSE  //not already set, keep all this boring stuff in
+	anchored = FALSE
+
+	mob_type = /mob/living/simple_animal/hostile/mining_drone  //what we're making
+	mob_name = "a minebot"
+	death = FALSE
+	roundstart = FALSE
+	short_desc = "You are a minebot, an expendable robot that supplies the station with ores."  //Printed in the spawner menu
+	flavour_text = {"
+	You are a minebot, a cheap and expendable mining companion, the station's crew have activated you so that you will assist their mining operation.
+	Clicking while in combat mode will fire the PKA, a rock-breaking tool that will not target sentient creatures.
+	Moving while in collect mode will scoop up ores; release them by pressing the EJECT button.
+	Clicking marker beacons in collect mode will add them to your inventory
+	You are immune to ash storms but be wary of lava pools.
+	Ask a nearby crewmember for upgrades to expand your potential.
+
+	Directives:
+	1. Do not harm sentient life.  Your weapon is automatically disabled when targeting sentient life.
+	2. Follow instructions issued by Nanotrasen crewmembers, except where that would be injurious to sentient life.
+	3. Wherever possible, mine ores from the rock walls around you and deposit those ores into an Ore Redemption Machine.
+	"}  //Printed when we spawn
```

Comment:
Try to use `\` at the end of a line instead of using curly brackets

Reply:
Curly braces are a completely valid way of writing long multiline strings

Reply:
this is the same syntax used by swarmers.
I copy pasted their spawner definition and adjusted it to fit.

might want to look at theirs actually, because it uses a lot of the <b> tag, which isn't parsed in the spawners menu, when I was testing

Reply:
bold tag
<b > and </b >

Reply:
> this is the same syntax used by swarmers.

It is **only** used by swarmers, every other mob uses \

Reply:
> Curly braces are a completely valid way of writing long multiline strings

It's "valid" but it is very rarely used in DM outside of defines.
Code standardisation is important

Reply:
okay, I will change that